### ISSUE TITLE
Non-integrated environments not working

### ISSUE THREAD

**Issue by insomniacc** [ironmansoftware/powershell-universal#3122]
### Version

4.2.12

### Severity

Critical

### Steps to Reproduce

New install, PO is currently being processed for a license.
Installed on a windows VM in a very locked down network environment, (vm uses a proxy for outbound traffic which is VERY restrictive).

I've created a simple "hello world" script and it runs fine in the integrated environment.
Trying to run in this in any other environment just hangs and eventually times out (ps7 or 5.1)
I believe this is due to outbound CRL checking requirements mentioned here: https://forums.ironmansoftware.com/t/non-integrated-environments-are-not-working/10338/6

### Expected behavior

```console
Script runs in seconds and completes successfully
```

### Actual behavior

```console
Script hangs, pwsh process is noticed as laucnhed but soon after ends, PSU reports the job is still running and then times out with the following error:

"Error executing job: PowerShell did not respond in a timely fashion. PowerShell Universal is currently waiting 30 seconds for PowerShell to start running the script.
You can increase the timeout by setting the 'Job Handshake Timeout' setting in the Settings \ General \ Automation page.
If you are still experiencing issues, you may need to consider increasing the resources available on your system or take advantage of the Random Delay parameter for schedules to stagger jobs."
```

### Additional Environment data

4.2.12
Installed as a service
Windows VM

Whitelisting on our proxy is only allowed via hostname.

**Comment by adamdriscoll** [comment]
This issue has been mentioned on **Ironman Software Forums**. There might be relevant details there:

https://forums.ironmansoftware.com/t/non-integrated-environments-are-not-working/10338/7

**Comment by adamdriscoll** [comment]
Here's a related KB article: https://support.ironmansoftware.com/portal/en/kb/articles/kb0015-offline-psu-server-has-slow-dashboards Let me know if that helps.

**Comment by insomniacc** [comment]
Hmm, I can try that, but my server isnt offline per say it's just behind a restrictive proxy but I can get anything I need whitelisted as long as its a domain name.
Both digicert & ctldl.windowsupdate.com are whitelisted and accessible, I'm able to get to that disallowedcertstl.cab file for example. 
Let me update that setting anyway and test and I'll get back to you!

**Comment by insomniacc** [comment]
@adamdriscoll - Just tried the above but the same issue is occurring!

**Comment by insomniacc** [comment]
Really hitting a brick wall with this one, not sure what else to try. I've confirmed that my proxy is configured under system which is what the service is running as, everything else thats impacted by the proxy (e.g OIDC Auth) responds and is impacted when i enable/disable the proxy, so I know its configured okay.
I've also tried running a wireshark packet capture to work out if there's any other outbound attempts being made that are getting blocked and I'm struggling to see anything of relevance there too. Still waiting on my org to process the PO too, not sure why it's taking so long but as soon as the license comes through I'll raise this as a support ticket.

**Comment by adamdriscoll** [comment]
One thing I would suggest would be to capture a memory dump of the PS process so we can see why it is hanging on startup.

**Comment by insomniacc** [comment]
Just tried but honestly I can't make heads or tails of it or find anything useful in the dump file.
![image](https://github.com/ironmansoftware/issues/assets/10046621/bd9331df-752c-440e-9676-9549f42cc8cc)
![image](https://github.com/ironmansoftware/issues/assets/10046621/d14f88fa-f580-4af5-9a6f-8ee8e92ff0ac)
I can see this in the stack, assuming that's the bit that's calling pwsh.exe but I'm not sure how else to interact with the dump file to get anything useful out of it.

I've confirmed the PO has finally been processed on our end and I've got my PSU license now, would it be possible to link up on this issue at some point? this really is a show stopper for me 😟

**Comment by insomniacc** [comment]
I ran an !analyze -v

```

*** WARNING: Unable to verify checksum for host.dll

KEY_VALUES_STRING: 1

    Key  : Analysis.CPU.mSec
    Value: 4499

    Key  : Analysis.Elapsed.mSec
    Value: 6734

    Key  : Analysis.IO.Other.Mb
    Value: 0

    Key  : Analysis.IO.Read.Mb
    Value: 1

    Key  : Analysis.IO.Write.Mb
    Value: 0

    Key  : Analysis.Init.CPU.mSec
    Value: 1187

    Key  : Analysis.Init.Elapsed.mSec
    Value: 92637

    Key  : Analysis.Memory.CommitPeak.Mb
    Value: 306

    Key  : CLR.Engine
    Value: CORECLR

    Key  : CLR.Version
    Value: 8.0.123.58001

    Key  : Failure.Bucket
    Value: BREAKPOINT_80000003_coreclr.dll!Thread::DoAppropriateWaitWorker

    Key  : Failure.Hash
    Value: {2dbd2c7d-1f72-d894-343d-110a7c82a3a2}

    Key  : Failure.Source.FileLine
    Value: 3467

    Key  : Failure.Source.FilePath
    Value: D:\a\_work\1\s\src\coreclr\vm\threads.cpp

    Key  : Failure.Source.SourceServerCommand
    Value: raw.githubusercontent.com/dotnet/runtime/bf5e279d9239bfef5bb1b8d6212f1b971c434606/src/coreclr/vm/threads.cpp

    Key  : Timeline.OS.Boot.DeltaSec
    Value: 8604

    Key  : Timeline.Process.Start.DeltaSec
    Value: 31

    Key  : WER.OS.Branch
    Value: fe_release

    Key  : WER.OS.Version
    Value: 10.0.20348.1

    Key  : WER.Process.Version
    Value: 7.4.1.500

FAULTING_THREAD:  00000310

PROCESS_NAME:  pwsh.exe

ERROR_CODE: (NTSTATUS) 0x80000003 - {EXCEPTION}  Breakpoint  A breakpoint has been reached.

STACK_COMMAND:  ~0s; .ecxr ; kb

FAULTING_SOURCE_LINE:  D:\a\_work\1\s\src\coreclr\vm\threads.cpp

FAULTING_SOURCE_FILE:  D:\a\_work\1\s\src\coreclr\vm\threads.cpp

FAULTING_SOURCE_LINE_NUMBER:  3467

FAULTING_SOURCE_SRV_COMMAND:  https://raw.githubusercontent.com/dotnet/runtime/bf5e279d9239bfef5bb1b8d6212f1b971c434606/src/coreclr/vm/threads.cpp

FAULTING_SOURCE_CODE:  
No source found for 'D:\a\_work\1\s\src\coreclr\vm\threads.cpp'

```

Is this relevant? I can confirm that raw.githubusercontent.com is blocked on this VM - I can try get that unblocked if you think this might be the issue?

**Comment by adamdriscoll** [comment]
@insomniacc - Can you send me an email at adam[at]ironmansoftware.com? I can give you a dropbox link to upload this dump and I will take a look

**Comment by insomniacc** [comment]
As an update to this issue. It was my proxy blocking gprc connections to localhost.
I got this working by adding the following environment variables to my system.

http_proxy = "http://proxyaddress:port"
https_proxy = "http://proxyaddress:port"
no_proxy = "127.0.0.1,localhost"

Note that case-sensitivity is important here, different libraries look for different case for these vars (theres no set standard although the mordern accepted defacto is lowercase, if in doubt, always use this). I had limited results when i first defined them in upper case and couldnt figure out what was going on.
Lowercase everything works as expected.
